homeassistant/components/frontier_silicon/media_player.py

```diff
@@ -175,43 +184,67 @@ def volume_level(self):
 
     async def async_update(self):
         """Get the latest date and update device state."""
-        fs_device = self.fs_device
-
-        if not self._name:
-            self._name = await fs_device.get_friendly_name()
-
-        if not self._source_list:
-            self._source_list = await fs_device.get_mode_list()
-
-        # The API seems to include 'zero' in the number of steps (e.g. if the range is
-        # 0-40 then get_volume_steps returns 41) subtract one to get the max volume.
-        # If call to get_volume fails set to 0 and try again next time.
-        if not self._max_volume:
-            self._max_volume = int(await fs_device.get_volume_steps() or 1) - 1
-
-        if await fs_device.get_power():
-            status = await fs_device.get_play_status()
-            self._state = {
-                "playing": STATE_PLAYING,
-                "paused": STATE_PAUSED,
-                "stopped": STATE_IDLE,
-                "unknown": STATE_UNKNOWN,
-                None: STATE_IDLE,
-            }.get(status, STATE_UNKNOWN)
+        afsapi = self.fs_device
+        try:
+            if await afsapi.get_power():
+                status = await afsapi.get_play_status()
+                self._state = {
+                    PlayState.PLAYING: STATE_PLAYING,
+                    PlayState.PAUSED: STATE_PAUSED,
+                    PlayState.STOPPED: STATE_IDLE,
+                    PlayState.LOADING: STATE_OPENING,
+                    None: STATE_IDLE,
+                }.get(status, STATE_UNKNOWN)
+            else:
+                self._state = STATE_OFF
+        except FSConnectionError:
+            if self._attr_available:
+                _LOGGER.warning(
+                    "Could not connect to %s. Did it go offline?",
+                    self._name or afsapi.webfsapi_endpoint,
+                )
+                self._state = STATE_UNAVAILABLE
+                self._attr_available = False
         else:
-            self._state = STATE_OFF
-
-        if self._state != STATE_OFF:
-            info_name = await fs_device.get_play_name()
-            info_text = await fs_device.get_play_text()
+            if not self._attr_available:
+                _LOGGER.info(
+                    "Reconnected to %s",
+                    self._name or afsapi.webfsapi_endpoint,
+                )
+
+                self._attr_available = True
+            if not self._name:
+                self._name = await afsapi.get_friendly_name()
+
+            if not self._attr_unique_id:
+                try:
+                    self._attr_unique_id = await afsapi.get_radio_id()
+                except FSApiException:
+                    self._attr_unique_id = self._attr_name
```

Comment:
A name is user-customizable and cannot be used as a unique ID.

Should the unique ID addition be in this PR? It seems not to be related to the version bump.

(Most of this refactoring seems to be unrelated).

Reply:
Removed for now. In a future PR I'll replace the fallback to throw a `ConfigEntryNotReady` instead.